nptdms/tdms.py

```diff
@@ -228,6 +232,39 @@ def _ipython_key_completions_(self):
         """
         return list(self._groups.keys())
 
+    def __eq__(self, other):
+        """ Compare two TdmsFile objects by properties, group names, channel names and channel values
+        """
+        equals = []
```

Comment:
Rather than building up a list of equality checks and then testing them all at the end, you could just return early the first time you find something that's not equal, then return True if you get to the end of the method without finding a difference.